pyFAI/ext/preproc.pyx

```diff
@@ -62,7 +63,7 @@ cdef floating[::1]c1_preproc(floating[::1] data,
                              floating delta_dummy=0.0,
                              bint check_dummy=False,
                              floating normalization_factor=1.0,
-                             ) nogil:
+                             ) with gil:
```

Comment:
I think the GIL is not even released when these functions are called below, so you can just leave out the GIL declaration completely now (unless it's called elsewhere as well).